File: src/com/fenghuo/notes/AddNoteActivity.java
package com.fenghuo.notes;

import android.app.Activity;
import android.content.Context;
import android.os.Bundle;
import android.view.View;
import android.view.ViewGroup;
import android.view.inputmethod.InputMethodManager;
import android.widget.Button;

import com.fenghuo.LineEditText;
import com.fenghuo.notes.bean.Note;
import com.fenghuo.notes.db.DBNoteHelper;
import com.fenghuo.notes.db.PreferenceHelper;

import java.util.Random;
import java.util.Timer;
import java.util.TimerTask;

public class AddNoteActivity extends Activity implements View.OnClickListener {

    private Button btn_back;
    private Button btn_save;
    private ViewGroup mScrollView;
    private LineEditText et_content;
    private DBNoteHelper noteHelper;// 数据库
    private PreferenceHelper helper;// 用于存储草稿
    private Random random;// 用于分配随机图片
    private boolean isSaved = false;
    private Note currentNote;

    @Override
    protected void onCreate(Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_add_thing);

        findviews();

        random = new Random();
        helper = new PreferenceHelper(AddNoteActivity.this);
        noteHelper = new DBNoteHelper(AddNoteActivity.this);

        btn_back.setOnClickListener(this);
        btn_save.setOnClickListener(this);
        // 上次没有保存的草稿
        et_content.setText(helper.getlast());
        et_content.setLineDown(et_content.getPaddingTop() - 2);
        // 自动弹出键盘
        Timer timer = new Timer();
        timer.schedule(new TimerTask() {
            @Override
            public void run() {

                InputMethodManager manager = (InputMethodManager) et_content
                        .getContext().getSystemService(
                                Context.INPUT_METHOD_SERVICE);
                manager.showSoftInput(et_content, 0);
            }
        }, 500);

    }

    private void findviews() {
        btn_back = (Button) findViewById(R.id.btn_back_addthi);
        btn_save = (Button) findViewById(R.id.btn_save_addthi);
        mScrollView = (ViewGroup) findViewById(R.id.scroll_view);
        et_content = (LineEditText) findViewById(R.id.et_text_addthi);
    }

    @Override
    protected void onDestroy() {

        if (!isSaved)
            helper.savelase(et_content.getText() + "");
        super.onDestroy();
    }

    // 返回 /保存 按钮事件
    @Override
    public void onClick(View arg0) {

        String content = et_content.getText().toString().trim();

        switch (arg0.getId()) {
            case R.id.btn_back_addthi:
                if (content.length() > 0)
                    helper.savelase(content);
                finish();
                break;
            case R.id.btn_save_addthi:
                currentNote = new Note(-1, random.nextInt(5), content, noteHelper.GetDate(), 0);
                noteHelper.Add(currentNote);
                helper.savelase("");// 如果是插入一条新记事 则需要清空last
                noteHelper.Desdroy();
                isSaved = true;
                finish();
                break;
            default:
                break;
        }
    }

}

File: src/com/fenghuo/notes/AddNoteActivity.kt
package com.fenghuo.notes

import android.app.Activity
import android.content.Context
import android.os.Bundle
import android.view.View
import android.view.ViewGroup
import android.view.inputmethod.InputMethodManager
import android.widget.Button
import com.fenghuo.LineEditText
import com.fenghuo.notes.bean.Note
import com.fenghuo.notes.db.DBNoteHelper
import com.fenghuo.notes.db.PreferenceHelper
import java.util.*

class AddNoteActivity : Activity(), View.OnClickListener {

    private var btn_back: Button? = null
    private var btn_save: Button? = null
    private var mScrollView: ViewGroup? = null
    private var et_content: LineEditText? = null
    private var noteHelper: DBNoteHelper? = null// 数据库
    private var helper: PreferenceHelper? = null// 用于存储草稿
    private var random: Random? = null// 用于分配随机图片
    private var isSaved = false
    private var currentNote: Note? = null

    override fun onCreate(savedInstanceState: Bundle?) {

        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_add_thing)

        btn_back = findViewById(R.id.btn_back_addthi)
        btn_save = findViewById(R.id.btn_save_addthi)
        mScrollView = findViewById(R.id.scroll_view)
        et_content = findViewById(R.id.et_text_addthi)

        random = Random()
        helper = PreferenceHelper(this@AddNoteActivity)
        noteHelper = DBNoteHelper(this@AddNoteActivity)

        btn_back!!.setOnClickListener(this)
        btn_save!!.setOnClickListener(this)
        // 上次没有保存的草稿
        et_content!!.setText(helper!!.last)
        et_content!!.setLineDown(et_content!!.paddingTop - 2)
        // 自动弹出键盘
        et_content!!.postDelayed({
            val manager = et_content!!
                    .context.getSystemService(
                    Context.INPUT_METHOD_SERVICE) as InputMethodManager
            manager.showSoftInput(et_content, 0)
        }, 500)
    }

    override fun onSaveInstanceState(outState: Bundle?) {
        super.onSaveInstanceState(outState)
        outState?.putString("saved_txt", et_content?.text.toString())
    }

    override fun onRestoreInstanceState(savedInstanceState: Bundle?) {
        super.onRestoreInstanceState(savedInstanceState)
        savedInstanceState?.getString("saved_txt")?.let { et_content?.setText(it) }
    }

    override fun onStop() {
        super.onStop()
        if (!isSaved) {
            helper!!.saveLast(et_content!!.text.toString() + "")
            isSaved = true
        }
    }

    // 返回 /保存 按钮事件
    override fun onClick(arg0: View) {

        val content = et_content!!.text.toString().trim { it <= ' ' }

        when (arg0.id) {
            R.id.btn_back_addthi -> {
                if (content.isNotEmpty())
                    helper!!.saveLast(content)
                finish()
            }
            R.id.btn_save_addthi -> {
                currentNote = Note(-1, random!!.nextInt(5), content, noteHelper!!.GetDate(), 0)
                noteHelper!!.Add(currentNote!!)
                helper!!.saveLast("")// 如果是插入一条新记事 则需要清空last
                noteHelper!!.Desdroy()
                isSaved = true
                finish()
            }
            else -> {
            }
        }
    }

}
